proselint/checks/__init__.py

```diff
@@ -1 +1,394 @@
-"""All the checks are organized into modules and places here."""
+"""All the checks are organized into modules and placed in subdirectories.
+
+This file contains:
+- contained function-set called by linter
+- _check-functions used by the checks
+
+This submodule gets used for multiprocessing!
+All custom code is contained in here.
+
+"""
+
+from __future__ import annotations
+
+import functools
+import importlib
+import re
+from enum import Enum
+from typing import Callable
+from typing import Optional
+from typing import TypeAlias
+
+from proselint.logger import log
+
+ResultCheck: TypeAlias = tuple[int, int, str, str, Optional[str]]
+# content: start_pos, end_pos, check_name, message, replacement)
+# NOTE1: NewType() is too strict here
+# NOTE2: py312 can use -> type ResultCheck = tuple[int, int, str, str, Optional[str]]
+
+
+###############################################################################
+# Check-Interface used by linter ##############################################
+###############################################################################
```

Comment:
as unexperienced users might browse through this file while writing custom checks I would like to keep those clear sections. it would also be an option to seperate FNs into sub-files.